client/argocdServer/application/Application.go

```diff
@@ -790,23 +791,40 @@ func getRolloutPodTemplateHash(pod map[string]interface{}) string {
 func buildPodMetadataFromPod(resp *v1alpha1.ApplicationTree, podManifests []map[string]interface{}, newPodNames map[string]bool) (podMetadata []*PodMetadata) {
 	containerMapping := make(map[string][]*string)
 	initContainerMapping := make(map[string][]*string)
+	var parentWorkflow []string
 	for _, pod := range podManifests {
 		containerMapping[getResourceName(pod)] = getPodContainers(pod)
 	}
 
 	for _, pod := range podManifests {
 		initContainerMapping[getResourceName(pod)] = getPodInitContainers(pod)
 	}
+
+	for _, node := range resp.Nodes {
+		if node.Kind == "Workflow" {
+			parentWorkflow = append(parentWorkflow, node.Name)
+		}
+	}
+
 	for _, node := range resp.Nodes {
 		if node.Kind == "Pod" {
-			isNew := newPodNames[node.Name]
+			isNew := newPodNames[node.Name] || contains(parentWorkflow, node.Name)
```

Comment:
contains should be moved out as newPodNames is provided as input argument, we should use this logic in the function which populates newPodNames